packages/node-http-handler/src/stream-collector/index.ts

```diff
@@ -3,8 +3,12 @@ import { Readable } from "stream";
 
 import { Collector } from "./collector";
 
-export const streamCollector: StreamCollector = (stream: Readable): Promise<Uint8Array> =>
-  new Promise((resolve, reject) => {
+export const streamCollector: StreamCollector = (stream: Readable | ReadableStream): Promise<Uint8Array> => {
+  if (isReadableStreamInstance(stream)) {
+    // web stream in Node.js indicates user has overridden requestHandler with FetchHttpHandler.
```

Comment:
This stream collector is specific to Node.js. Does this mean user has passed Node.js WebStream in body?

```suggestion
  if (isReadableStreamInstance(stream)) {
    // Web stream API in Node.js
```

If that's the case, the type on line 6 should be
```js
import { ReadableStream } from "stream/web";
```

Reply:
```ts
import { ReadableStream } from "stream/web";
```
I don't think we can statically import this because of supporting Node.js 16.

```ts
import type ...
```
This might be possible? our TS types for Node.js 16 lib don't seem to support this right now, and I refreshed my installation today.

Reply:
> our TS types for Node.js 16 lib don't seem to support this right now

We need to bump `@types/node` for `@smithy/*` deps just like it's being done in https://github.com/aws/aws-sdk-js-v3/pull/6047. Let's do that first before merging this change?